Framework/Core/include/Framework/DeviceControl.h

```diff
@@ -12,21 +12,30 @@
 
 #include <map>
 #include <string>
+#include "Framework/LogParsingHelpers.h"
 
 namespace o2 {
 namespace framework {
 
-// Controller state for the Device. This is useful for both GUI and
-// batch operations of the system. Whenever something external to the
-// device wants to modify it, it should be registered here and it will
-// be acted on in the subsequent state update.
+/// Controller state for the Device. This is useful for both GUI and batch
+/// operations of the system. Whenever something external to the device wants
+/// to modify it, it should be registered here and it will be acted on in the
+/// subsequent state update.
 struct DeviceControl {
-  bool stopped; // whether the device should start in STOP
-  bool quiet; // wether we should be capturing device output.
-  char logFilter[256] = {0};  // Lines in the log should match this to be displayed
-  char logStartTrigger[256] = {0}; // Start printing log with the last occurence of this
-  char logStopTrigger[256] = {0}; // Stop producing log with the first occurrence of this after the start
-  std::map<std::string, std::string> options; // Where the GUI should store the options it wants.
+  // whether the device should start in STOP
+  bool stopped = false;
+  /// wether we should be capturing device output.
+  bool quiet = false;
+  /// Minimum log level for messages to appear 
+  LogParsingHelpers::LogLevel logLevel = LogParsingHelpers::LogLevel::Info;
+  /// Lines in the log should match this to be displayed  
+  char logFilter[256] = {0};
```

Comment:
not really touched by this PR, but lets define a constant holding the length. I found a couple of `asserts` checking the size of the `log*` members to be 256 in `FrameworkGUIDebugger.cxx`.

Also I noticed, in `FrameworkGUIDebugger.cxx` line 367
```
ImGui::InputText("Log filter", control.logFilter, sizeof(control.logFilter));
```
there is no check that the array is 0 terminated.

Reply:
I think it does not need to be 0 terminated. In the example the author of ImGUI however uses `IM_ARRAYSIZE`, so I will do the same.